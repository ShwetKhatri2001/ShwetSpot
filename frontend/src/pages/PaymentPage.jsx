import React, { useState } from 'react';
import { Button, Form } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import CheckoutBar from '../components/CheckoutBar';
import { savePaymentMethod } from '../redux/actions/cartActions';

const PaymentPage = ({ history, initialValues, onSubmit }) => {
  const cart = useSelector(state => state.cart);
  const { shippingAddress } = cart;

  if (!shippingAddress) {
    history.push('/shipping');
  }

  const [paymentMethod, setPaymentMethod] = useState('PayGreen');

  const dispatch = useDispatch();

  const submitHandler = e => {
    if (e) e.preventDefault();
    dispatch(savePaymentMethod(paymentMethod));
    history.push('/placeorder');
  };

  return (
    <>
      <CheckoutBar step1 step2 step3 />
      <div className="justify-content-center align-items-center container bg-light p-3">
        <h1 className="pl-3">Payment Method</h1>
        <Form onSubmit={submitHandler} className="pr-3 pl-3">
          <Form.Group>
            <Form.Label>Select Method</Form.Label>
            <Form.Check
              type="radio"
              label="PayGreen"
              checked={paymentMethod === 'PayGreen'}
              value="PayGreen"
              onChange={e => setPaymentMethod(e.target.value)}
              className="mb-1"
            />
            <Form.Check
              type="radio"
              label="FoxCard"
              checked={paymentMethod === 'FoxCard'}
              value="FoxCard"
              onChange={e => setPaymentMethod(e.target.value)}
            />
          </Form.Group>

          <Button type="submit" variant="success" className="mb-2">
            Continue
          </Button>
        </Form>
      </div>
    </>
  );
};

export default PaymentPage;
