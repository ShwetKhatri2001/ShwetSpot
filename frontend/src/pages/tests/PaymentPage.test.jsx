import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import PaymentPage from '../PaymentPage';
import store from '../../redux/store';

describe('PaymentPage Component', () => {
  const MockPaymentPage = (
    <MemoryRouter>
      <Provider store={store}>
        <PaymentPage />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockPaymentPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
