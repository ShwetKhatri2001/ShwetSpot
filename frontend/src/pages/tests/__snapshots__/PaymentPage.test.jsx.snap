// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentPage Component should match the snapshot 1`] = `
Array [
  <div
    className="justify-content-center mb-4 nav"
    onKeyDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link active"
        data-rb-event-key="/login"
        href="/login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Sign In
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/shipping"
        href="/shipping"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Shipping
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/payment"
        href="/payment"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Payment
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-disabled={true}
        className="nav-link disabled"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={-1}
      >
        Place Order
      </a>
    </div>
  </div>,
  <div
    className="justify-content-center align-items-center container bg-light p-3"
  >
    <h1
      className="pl-3"
    >
      Payment Method
    </h1>
    <form
      className="pr-3 pl-3"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Select Method
        </label>
        <div
          className="mb-1 form-check"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="PayGreen"
          />
          <label
            className="form-check-label"
            title=""
          >
            PayGreen
          </label>
        </div>
        <div
          className="form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="FoxCard"
          />
          <label
            className="form-check-label"
            title=""
          >
            FoxCard
          </label>
        </div>
      </div>
      <button
        className="mb-2 btn btn-success"
        disabled={false}
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>,
]
`;
